resources/Microsoft.DotNet.Dsc/Microsoft.DotNet.Dsc.psm1

```diff
@@ -262,24 +273,45 @@ function Update-DotNetToolPackage
         [string] $PackageId,
         [string] $Version,
         [bool]   $PreRelease,
-        [string] $ToolPath,
-        [bool]   $Exist
+        [string] $ToolPathDirectory,
+        [bool]   $Exist,
+        [switch] $Downgrade
```

Comment:
I don't think a downgrade switch is needed. If the downgrade option is not supported by the dotnet tool, we could just try uninstalling and reinstalling the correct version? I would opt for that option in case a user doesn't know what version is currently on the machine.

Reply:
Got it! Please see latest commit. I have opted to use a helper method ReInstall and just call Uninstall/Install to simplify things. You got 4 scenarios now:

- ReInstall
- Upgrade
- Install
- Uninstall